### ISSUE TITLE
Small pre-trained models

### ISSUE THREAD

**Issue by sara-dahl** [davidsandberg/facenet#146]
@davidsandberg:

I wonder if you could also provide some smaller pre-trained models, e.g., for the nn4_small2_v1. 

I understand that you are just looking for a better accuracy. However, in some applications, the simplicity is much more important. For example, you cannot load and run the current model on a mobile application!

Please note that one of the beauties of the FaceNet that the authors brag about is the small 128-bytes embedding, which has become 1792-bytes in the current Inception-ResNet model.

Thanks a lot for considering this.
 
Sara

**Comment by davidsandberg** [comment]
Hi @sara-dahl!
Currently I have no plans to provide a pre-trained model based on nn4_small2_v1. But all the code for training is in the repo so it shouldn't be too difficult to run the training. But it would anyway require some tuning of hyper-parameters (weight decay etc).
However I still think that reducing the size of the embedding is very interesting. I know that @dleonsanchez has done some excellent work reducing the embedding size from 1792 down to 128 using PCA. It would be interesting to investigate if one could introduce a bottleneck (128 wide fully connected layer) in inception_resnet_v1 that does essentially the same thing as a PCA.

**Comment by zhenglaizhang** [comment]
@sara-dahl hi, have you got some idea about smaller model of facenet? I am currently want to test facenet in android, but currently model file is a bit larger. thanks in advance!

**Comment by dxl520** [comment]
@zhenglaizhang hi，Can you find a face detection model suitable for mobile phones now?thank you！
